diff --git a/node_modules/@netlify/build/lib/types/config/functions.d.ts b/node_modules/@netlify/build/lib/types/config/functions.d.ts
index 273066d..e56e320 100644
--- a/node_modules/@netlify/build/lib/types/config/functions.d.ts
+++ b/node_modules/@netlify/build/lib/types/config/functions.d.ts
@@ -4,6 +4,11 @@ type FunctionsObject = {
      * a list of additional paths to include in the function bundle. Although our build system includes statically referenced files (like `import * from "./some-file.js"`) by default, `included_files` lets you specify additional files or directories and reference them dynamically in function code. You can use `*` to match any character or prefix an entry with `!` to exclude files. Paths are relative to the [base directory](https://docs.netlify.com/configure-builds/get-started/#definitions-1).
      */
     included_files?: string[];
+
+    /**
+     * [patched] allow to set included files base
+     */
+    included_files_base_path?: string
 } & ({
     /**
      * the function bundling method used in [`@netlify/zip-it-and-ship-it`](https://github.com/netlify/zip-it-and-ship-it).
